sway-ir/src/verify.rs

```diff
@@ -389,11 +389,24 @@ impl<'a, 'eng> InstructionVerifier<'a, 'eng> {
             .get_type(self.context)
             .ok_or(IrError::VerifyBinaryOpIncorrectArgType)?;
 
-        if !arg1_ty.is_ptr(self.context)
-            || !arg2_ty.is_ptr(self.context)
-            || !result_ty.is_ptr(self.context)
-        {
-            return Err(IrError::VerifyBinaryOpIncorrectArgType);
+        match op {
+            // Shifts rhs are 64 bits
+            BinaryOpKind::Lsh | BinaryOpKind::Rsh => {
+                if !arg1_ty.is_ptr(self.context)
+                    || !arg2_ty.is_uint64(self.context)
+                    || !result_ty.is_ptr(self.context)
+                {
+                    return Err(IrError::VerifyBinaryOpIncorrectArgType);
+                }
+            }
+            _ => {
```

Comment:
You should explicit the other ops so we don't introduce a bug if a new one is added.

Reply:
Good catch. Done.